src/components/AddItemForm/AddItemForm.js

```diff
@@ -19,8 +28,55 @@ const AddItemForm = ({ listId }) => {
   const handleSubmit = async (event) => {
     event.preventDefault();
 
+    // check if itemName already exists in Firestore
+    try {
+      // get array of listItems from Firestore
+      await db
+        .collection(`lists/${listId}/items`)
+        .get()
+        .then(async (querySnapshot) => {
+          // create array (dbItemArray) by returning itemNames from Firestore response (querySnapshot)
+          let dbItemArray = querySnapshot.docs.map((doc) => {
+            return doc.data().itemName;
+          });
+
+          const duplicateResult = isItemDuplicate(
+            formValues.itemName,
+            dbItemArray,
+          );
+
+          // if item exists, show error message, otherwise, continue with adding to database
+          if (duplicateResult === true) {
+            setErrorMessage('Item already exists in Shopping List');
+          } else {
+            addItemToDatabase();
+          }
+        });
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const isItemDuplicate = (item, array) => {
+    const itemToCompare = normalizeInput(item);
+    const arrayToCompare = array.map((dbItem) => normalizeInput(dbItem));
+
+    // if .indexOf returns -1, the item does not exist in array, aka it's not a duplicate, aka false
+    return arrayToCompare.indexOf(itemToCompare) === -1 ? false : true;
+  };
+
+  // used when comparing entered item and array of db items for duplicates
+  const normalizeInput = (item) => {
+    return item
+      .toLowerCase()
+      .replace(/\W/g, '') // remove non-word characters aka remove punctuation
+      .replace(/ /g, '') // remove spaces
+      .trim(); // remove leading and trailing whitespace (trim must be last)
+  };
+
+  const addItemToDatabase = async () => {
```

Comment:
I like the choice here to separate this out into another function. 👍

Reply:
I concur. 👏🏽

Reply:
Yes! This is an excellent improvement in readability. Well done 👏🏽